tests/microos/networkmanager.pm

```diff
@@ -40,6 +40,7 @@ sub restore_config {
     script_run('rm -f /etc/NetworkManager/conf.d/00-use-dnsmasq.conf');
     script_run('unlink /etc/resolv.conf');
     systemctl('restart NetworkManager');
+    script_retry('systemctl restart NetworkManager-wait-online.service', retry => 3, delay => 20);
```

Comment:
Just out of curiosity, have you checked `nm-online`? You can set timeout over there and most likely you might receive some useful output in case of failure.

Reply:
In this case, I believe that relying on `NetworkManager-wait-online.service` as it executes `nm-online` is not the right check to make sure that NM has started

Reply:
I checked NM status, it started successfully.  But online service cannot be started.